test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ModelBinderExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    public static class ModelBinderExtensions
+    {
+        public static async Task<ModelBindingResult> BindModelResultAsync(
```

Comment:
Looks like this has the potential to undo some gains main in this PR. Could we instead push the new pattern all the way down?